h2o-mapreduce-generic/src/main/java/water/hadoop/FileSystemBasedClouding.java

```diff
@@ -0,0 +1,140 @@
+package water.hadoop;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.*;
+
+import java.io.*;
+import java.net.InetAddress;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FileSystemBasedClouding extends AbstractClouding {
+
+  private InetAddress _self_ip;
+  private int _self_port = -1;
+  
+  private int _n;
+  private Path _path;
+  private FileSystem _fs;
+
+  private final Map<String, URI> _nodes = new HashMap<>();
+
+  void init(Configuration conf) throws IOException {
+    _n = conf.getInt(h2omapper.H2O_CLOUD_SIZE_KEY, -1);
+    _path = new Path(conf.get(h2omapper.H2O_CLOUDING_DIR_KEY));
+    _fs = FileSystem.get(conf);
+  }
+
+  @Override
+  public void notifyAboutCloudSize(InetAddress ip, int port, InetAddress leaderIp, int leaderPort, int size) {
+    if (size != _n)
+      return;
+    Path path = toNodePath(ip, port, "leader");
+    try {
+      writeFile(path, h2oUri(leaderIp, leaderPort).toString());
+    } catch (IOException e) {
+      e.printStackTrace();
+      exit(162);
+    }
+  }
+
+  @Override
+  public void notifyAboutEmbeddedWebServerIpPort(InetAddress ip, int port) {
```

Comment:
maybe also remove the huge TODO in NetworkInit ;)

Reply:
agreed, it is deprecated at this point